### ISSUE TITLE
Segmentation fault (core dump) in cascade protocol

### ISSUE THREAD

**Issue by luckyluke98** [sequence-toolbox/SeQUeNCe#166]
I get the following segmentation fault error while running the cascade.

Fatal Python error: Segmentation fault

Current thread 0x00007f09020e1000 (most recent call first):
  File "/home/lcky/.local/lib/python3.10/site-packages/sequence/qkd/cascade.py", line 484 in create_checksum_table
  File "/home/lcky/.local/lib/python3.10/site-packages/sequence/qkd/cascade.py", line 217 in pop
  File "/home/lcky/.local/lib/python3.10/site-packages/sequence/protocol.py", line 83 in _pop
  File "/home/lcky/.local/lib/python3.10/site-packages/sequence/qkd/BB84.py", line 388 in received_message
  File "/home/lcky/.local/lib/python3.10/site-packages/sequence/topology/node.py", line 615 in receive_message
  File "/home/lcky/.local/lib/python3.10/site-packages/sequence/kernel/process.py", line 31 in run
  File "/home/lcky/.local/lib/python3.10/site-packages/sequence/kernel/timeline.py", line 133 in run
  File "/mnt/c/Users/Vinci/Desktop/QKDSimulator/project/sim_ext.py", line 59 in sim
  File "/mnt/c/Users/Vinci/Desktop/QKDSimulator/project/sim_ext.py", line 91 in main
  File "/mnt/c/Users/Vinci/Desktop/QKDSimulator/project/sim_ext.py", line 116 in <module>

Extension modules: numpy.core._multiarray_umath, numpy.core._multiarray_tests, numpy.linalg._umath_linalg, numpy.fft._pocketfft_internal, numpy.random._common, numpy.random.bit_generator, numpy.random._bounded_integers, numpy.random._mt19937, numpy.random.mtrand, numpy.random._philox, numpy.random._pcg64, numpy.random._sfc64, numpy.random._generator, qutip.cy.spmatfuncs, scipy._lib._ccallback_c, matplotlib._c_internal_utils, PIL._imaging, matplotlib._path, kiwisolver._cext, scipy.sparse._sparsetools, _csparsetools, scipy.sparse._csparsetools, scipy.sparse.linalg._isolve._iterative, scipy.linalg._fblas, scipy.linalg._flapack, scipy.linalg._cythonized_array_utils, scipy.linalg._flinalg, scipy.linalg._solve_toeplitz, scipy.linalg._matfuncs_sqrtm_triu, scipy.linalg.cython_lapack, scipy.linalg.cython_blas, scipy.linalg._matfuncs_expm, scipy.linalg._decomp_update, scipy.sparse.linalg._dsolve._superlu, scipy.sparse.linalg._eigen.arpack._arpack, scipy.sparse.csgraph._tools, scipy.sparse.csgraph._shortest_path, scipy.sparse.csgraph._traversal, scipy.sparse.csgraph._min_spanning_tree, scipy.sparse.csgraph._flow, scipy.sparse.csgraph._matching, scipy.sparse.csgraph._reordering, qutip.cy.spmath, qutip.cy.spconvert, qutip.cy.ptrace, qutip.cy.sparse_utils, qutip.cy.interpolate, scipy.special._ufuncs_cxx, scipy.special._ufuncs, scipy.special._specfun, scipy.special._comb, scipy.special._ellip_harm_2, scipy.interpolate._fitpack, scipy.interpolate.dfitpack, scipy.optimize._minpack2, scipy.optimize._group_columns, scipy._lib.messagestream, scipy.optimize._trlib._trlib, numpy.linalg.lapack_lite, scipy.optimize._lbfgsb, _moduleTNC, scipy.optimize._moduleTNC, scipy.optimize._cobyla, scipy.optimize._slsqp, scipy.optimize._minpack, scipy.optimize._lsq.givens_elimination, scipy.optimize._zeros, scipy.optimize.__nnls, scipy.optimize._highs.cython.src._highs_wrapper, scipy.optimize._highs._highs_wrapper, scipy.optimize._highs.cython.src._highs_constants, scipy.optimize._highs._highs_constants, scipy.linalg._interpolative, scipy.optimize._bglu_dense, scipy.optimize._lsap, scipy.spatial._ckdtree, scipy.spatial._qhull, scipy.spatial._voronoi, scipy.spatial._distance_wrap, scipy.spatial._hausdorff, scipy.spatial.transform._rotation, scipy.optimize._direct, scipy.interpolate._bspl, scipy.interpolate._ppoly, scipy.interpolate.interpnd, scipy.interpolate._rbfinterp_pythran, scipy.interpolate._rgi_cython, qutip.cy.inter, qutip.cy.cqobjevo_factor, qutip.cy.cqobjevo, qutip.cy.graph_utils, matplotlib._image, scipy._lib._uarray._uarray, scipy.fftpack.convolve, scipy.integrate._odepack, scipy.integrate._quadpack, scipy.integrate._vode, scipy.integrate._dop, scipy.integrate._lsoda, qutip.cy.brtools, qutip.cy.stochastic, qutip.cy.mcsolve, qutip.cy.br_tensor, scipy.ndimage._nd_image, _ni_label, scipy.ndimage._ni_label, scipy.special.cython_special, scipy.stats._stats, scipy.stats.beta_ufunc, scipy.stats._boost.beta_ufunc, scipy.stats.binom_ufunc, scipy.stats._boost.binom_ufunc, scipy.stats.nbinom_ufunc, scipy.stats._boost.nbinom_ufunc, scipy.stats.hypergeom_ufunc, scipy.stats._boost.hypergeom_ufunc, scipy.stats.ncf_ufunc, scipy.stats._boost.ncf_ufunc, scipy.stats.ncx2_ufunc, scipy.stats._boost.ncx2_ufunc, scipy.stats.nct_ufunc, scipy.stats._boost.nct_ufunc, scipy.stats.skewnorm_ufunc, scipy.stats._boost.skewnorm_ufunc, scipy.stats.invgauss_ufunc, scipy.stats._boost.invgauss_ufunc, scipy.stats._biasedurn, scipy.stats._levy_stable.levyst, scipy.stats._stats_pythran, scipy.stats._statlib, scipy.stats._mvn, scipy.stats._sobol, scipy.stats._qmc_cy, scipy.stats._rcont.rcont (total: 134)
Segmentation fault (core dumped)

Basically what I'm going to do is the execution of BB84, with the cascade on it, for an infinite number of keys, using math.inf (as also from your example), after some times (about 2 hours) I get the previous error. I don't think it's related to an error in my code.
